src/reanimated2/animation/timing.ts

```diff
@@ -67,14 +67,20 @@ export const withTiming = function (
 
   if (userConfig?.easing) {
     const { easing } = userConfig;
-    if (!('__workletHash' in easing)) {
+    const isFunction: boolean = typeof easing === 'function';
+    const isWorklet: boolean = !!(
+      easing as unknown as { __workletHash: number }
+    )?.__workletHash;
+    const functionName = typeof easing === 'function' ? easing?.name : '';
+    const isBound = functionName.startsWith('bound');
```

Comment:
does it work correctly across different JS engines?

Reply:
According to documentation it does: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind